ListUsers Production Best Practices (#772)
openfga/openfga.dev: docs/content/interacting/relationship-queries.mdx

## ListUsers

### What Is It For?

The [ListUsers API](/api/service#/Relationship%20Queries/ListUsers) is an API endpoint that that returns all users of a given type that have a specified relationship with an object.

### When To Use?

Use the ListUsers API to get which users have a relation to a specific object. 

<ListUsersRequestViewer
  authorizationModelId="01HVMMBCMGZNT3SED4Z17ECXCA"
  objectType="document"
  objectId="planning"
  relation="viewer"
  userFilterType="user"
  expectedResults={{
    users: [
      { object: { type: "user", id: "anne" }}, 
      { object: { type: "user", id: "beth" }}
    ]
  }}
/>

### Caveats

ListUsers will return the results found within the time allotted (`listUsersDeadline`, default: `3s`) up to the maximum number of results configured (`listUsersMaxResults`, default: `1000`). See [Configuring the Server](../getting-started/setup-openfga/docker-setup.mdx#configuring-the-server)) for more on how to change the default configuration.

- If you set `listUsersDeadline` to `1s`, the server will spend at most 1 second finding results.
- If you set `listUsersMaxResults` to `10`, the server will return, at most, 10 objects.

> If the number of users of that type is high, you should set a high value for `listUsersDeadline`. If the number of objects of that type the user could have access to is high, you should set a high value for `listUsersMaxResults`.

Reviewer: ```suggestion
If the number of users matching that filter is high, you should set a high value for `listUsersDeadline`. If the number of users matching that filter that could have that relation with the object is high, you should set a high value for `listUsersMaxResults`.
```
